utilities/tools/module-indexes/module-index.tests.ps1

```diff
@@ -0,0 +1,747 @@
+                    $lineNumber = 2
+                    foreach ($item in $csvContent) {
+                        $item.ModuleName | Should -Match '^avm/ptn/.*' -Because "ModuleName should start with 'avm/ptn/'. In line #$lineNumber, this is invalid. The following values have been provided: ""$($item.ModuleName)"""
+                        $lineNumber++
+                    }
+                }
+
+                It 'Should have exactly 2 segments (after the prefix)' {
+                    $lineNumber = 2
+                    foreach ($item in $csvContent) {
+                        $truncatedModuleName = $item.ModuleName -replace 'avm/ptn/', ''
+                        $truncatedModuleName -split '/' | Should -HaveCount 2 -Because "ModuleName should have exactly 2 segments in line #$lineNumber"
+                        $lineNumber++
+                    }
+                }
+            } elseif ($CsvFilePath -match 'UtilityModules') {
+                It "Should start with 'avm/utl/'" {
+                    $lineNumber = 2
+                    foreach ($item in $csvContent) {
+                        $item.ModuleName | Should -Match '^avm/utl/.*' -Because "ModuleName should start with 'avm/utl/'. In line #$lineNumber, this is invalid. The following values have been provided: ""$($item.ModuleName)"""
+                        $lineNumber++
+                    }
+                }
+
+                It 'Should have exactly 2 segments (after the prefix)' {
+                    $lineNumber = 2
+                    foreach ($item in $csvContent) {
+                        $truncatedModuleName = $item.ModuleName -replace 'avm/utl/', ''
+                        $truncatedModuleName -split '/' | Should -HaveCount 2 -Because "ModuleName should have exactly 2 segments in line #$lineNumber"
+                        $lineNumber++
+                    }
+                }
+            }
+
+            It "All items in the CSV should be alphabetically ordered, based on ModuleName" {
+
+                # Set-ItResult -Skipped
+
+                # $moduleNames = $csvContent.ModuleName
+                # $sortedModuleNames = $moduleNames | Sort-Object -Stable -Culture "en-US"
+                # for ($i = 0; $i -lt $moduleNames.Length; $i++) {
+                #     $moduleNames[$i] | Should -Be $sortedModuleNames[$i] -Because "All items in the CSV should be alphabetically ordered, based on ModuleName"
+                # }
+
+                # $moduleNames = $csvContent.ModuleName
+                # $normalizedModuleNames = $moduleNames -replace '[/\-]', ''
+                # $sortedModuleNames = $normalizedModuleNames | Sort-Object -Stable -Culture "en-US"
+
+                # for ($i = 0; $i -lt $moduleNames.Length; $i++) {
+                #     $normalizedModuleNames[$i] | Should -Be $sortedModuleNames[$i] -Because "All items in the CSV should be alphabetically ordered, based on normalized ModuleName"
+                # }
```

Comment:
Leftover?

Reply:
Looks like it. :) I wanted to remove this entire test case for now, since as it turns out, Excel orders strings separated with dashes, slashes, etc. in a very weird way. I couldn't figure out an easy way how PowerShell could apply the exact same algorithm for sorting, and hence, there was always a few differences. As the solution used to render the content on the AVM website is also ordering the item alphabetically, this is just cosmetics, and hence not worth the effort.

Reply:
Or at least in case the forward slashes of Terraform, this was definitely the case. I may keep the Bicep solution for now.